src/lib/blockchain_snark/blockchain_transition.ml

```diff
@@ -238,13 +238,17 @@ module Make (T : Transaction_snark.Verification.S) = struct
           in
           (verification, {proving with value= ()})
         in
+        let r1cs use_dummy_key =
+          Tock.constraint_system ~exposing:Wrap.input
+            (Wrap.main ~use_dummy_key)
+        in
         Cached.Spec.create ~load ~name:"blockchain-snark wrap keys"
           ~autogen_path:Cache_dir.autogen_path
           ~manual_install_path:Cache_dir.manual_install_path
-          ~digest_input:
-            (Fn.compose Md5.to_hex Tock.R1CS_constraint_system.digest)
-          ~create_env:Tock.Keypair.generate
-          ~input:(Tock.constraint_system ~exposing:Wrap.input Wrap.main)
+          ~digest_input:(fun () ->
+            Tock.R1CS_constraint_system.digest (r1cs true) |> Md5.to_hex )
```

Comment:
I think you noticed this IRL yesterday, but leaving this comment here so we don't forget: this is wrong, and will prevent the cache from being updated correctly. We only want to use dummy keys when computing `constraint-system-digests`.